### PR TITLE
feat: Recent transaction to mvvm

### PR DESCRIPTION
**Pull request by PratyushSingh07**

### CODE REVIEW

**Comment by PratyushSingh07** [comment]
hey @jawidMuhammadi could please review this !!

File: app/src/main/java/org/mifos/mobile/utils/TransactionLoadingState.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.mifos.mobile.utils
+
+import org.mifos.mobile.models.Transaction
+
+
+sealed class TransactionLoadingState {
+    object ShowProgress : TransactionLoadingState()
+    object HideProgress : TransactionLoadingState()
```
Comment (jawidMuhammadi) [inline app/src/main/java/org/mifos/mobile/utils/TransactionLoadingState.kt]: You could combine both to `Loading`

File: app/src/main/java/org/mifos/mobile/utils/TransactionLoadingState.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package org.mifos.mobile.utils
+
+import org.mifos.mobile.models.Transaction
+
+
+sealed class TransactionLoadingState {
```
Comment (jawidMuhammadi) [inline app/src/main/java/org/mifos/mobile/utils/TransactionLoadingState.kt]: Rename to `TransactionUiState`

File: app/src/main/java/org/mifos/mobile/ui/fragments/RecentTransactionsFragment.kt [hunk 3]
```diff
@@ -55,18 +54,51 @@ class RecentTransactionsFragment : BaseFragment(), RecentTransactionsView, OnRef
...
+            when (state) {
+                is TransactionLoadingState.ShowProgress -> {
+                    showProgress()
+                }
+                is TransactionLoadingState.HideProgress -> {
+                    hideProgress()
+                }
+                is TransactionLoadingState.RecentTransactions -> {
+                    hideProgress()
+                    val transactions: List<Transaction> = state.transactions
```
Comment (jawidMuhammadi) [inline app/src/main/java/org/mifos/mobile/ui/fragments/RecentTransactionsFragment.kt]: Do we need to create the transactions variable here? What about ` showRecentTransactions(state.transactions)` ?

**Review by jawidMuhammadi — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobile/ui/fragments/RecentTransactionsFragment.kt (hunk 3 above)
Comment (PratyushSingh07) [reply app/src/main/java/org/mifos/mobile/ui/fragments/RecentTransactionsFragment.kt]: yes we dont need to . I will wrap it up in a single line then

File: app/src/main/java/org/mifos/mobile/utils/TransactionLoadingState.kt (hunk 2 above)
Comment (PratyushSingh07) [reply app/src/main/java/org/mifos/mobile/utils/TransactionLoadingState.kt]: done

File: app/src/main/java/org/mifos/mobile/utils/TransactionLoadingState.kt (hunk 1 above)
Comment (PratyushSingh07) [reply app/src/main/java/org/mifos/mobile/utils/TransactionLoadingState.kt]: done

File: app/src/main/java/org/mifos/mobile/ui/fragments/RecentTransactionsFragment.kt (hunk 3 above)
Comment (PratyushSingh07) [reply app/src/main/java/org/mifos/mobile/ui/fragments/RecentTransactionsFragment.kt]: hey @jawidMuhammadi , I made the necessary changes do review this while I work on other Open PRs

File: app/src/main/java/org/mifos/mobile/utils/TransactionUiState.kt [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+sealed class TransactionUiState {
+    data class Loading(val showProgress: Boolean) : TransactionUiState()
+    data class RecentTransactions(val transactions: List<Transaction>) : TransactionUiState()
+    data class LoadMoreTransactions(val transactions: List<Transaction>) : TransactionUiState()
+    object EmptyTransaction : TransactionUiState()
+    data class ErrorFetchingTransactions(val errorMessageResId: Int) : TransactionUiState()
+
+    companion object {
+        val SHOW_PROGRESS = Loading(true)
+        val HIDE_PROGRESS = Loading(false)
```
Comment (jawidMuhammadi) [inline app/src/main/java/org/mifos/mobile/utils/TransactionUiState.kt:15]: No need for these, just you need to know when it is loading. You are already hiding the loading view when the data is successful or failed: 
image

I hope it makes sense to you!

So just add:

`object Loading: TransactionUiState()`

**Review by jawidMuhammadi — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobile/viewModels/RecentTransactionViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+import io.reactivex.schedulers.Schedulers
+import org.mifos.mobile.R
+import org.mifos.mobile.api.DataManager
+import org.mifos.mobile.models.Page
+import org.mifos.mobile.models.Transaction
+import org.mifos.mobile.utils.TransactionUiState
+import javax.inject.Inject
+
+class RecentTransactionViewModel @Inject constructor(
+    private val dataManager: DataManager?
```
Comment (jawidMuhammadi) [inline app/src/main/java/org/mifos/mobile/viewModels/RecentTransactionViewModel.kt:17]: Repository? 😊 You might create one repository for all the Transaction related data.

**Review by jawidMuhammadi — CHANGES_REQUESTED** [review]

**Pull request closed without merging**